verificationtool/src/main/java/com/greenops/verificationtool/ingest/handling/PipelineVerificationHandlerImpl.java

```diff
@@ -0,0 +1,93 @@
+package com.greenops.verificationtool.ingest.handling;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.greenops.util.datamodel.event.Event;
+import com.greenops.util.datamodel.pipelinestatus.FailedStep;
+import com.greenops.util.datamodel.pipelinestatus.PipelineStatus;
+import com.greenops.verificationtool.datamodel.verification.DAG;
+import com.greenops.verificationtool.ingest.apiclient.workflowtrigger.WorkflowTriggerApi;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+public class PipelineVerificationHandlerImpl implements PipelineVerificationHandler {
+    private final WorkflowTriggerApi workflowTriggerApi;
+    private final ObjectMapper objectMapper;
+
+    @Autowired
+    public PipelineVerificationHandlerImpl(WorkflowTriggerApi workflowTriggerApi, @Qualifier("objectMapper") ObjectMapper objectMapper){
+        this.workflowTriggerApi = workflowTriggerApi;
+        this.objectMapper = objectMapper;
+    }
+
+    private PipelineStatus getPipelineStatus(Event event){
+        String pipelineStatusStr = this.workflowTriggerApi.getPipelineStatus(event.getOrgName(), event.getPipelineName(), event.getTeamName());
+        System.out.println("Pipeline Status: " + pipelineStatusStr);
+        try {
+            PipelineStatus pipelineStatus = this.objectMapper.readValue(pipelineStatusStr, PipelineStatus.class);
+            return pipelineStatus;
+        } catch (JsonProcessingException e){
+            throw new RuntimeException("Unable to serialize pipeline status into PipelineStatus Object " + e);
+        }
+    }
+
+    @Override
+    public Boolean verify(Event event, DAG dag){
+        PipelineStatus pipelineStatus = getPipelineStatus(event);
+        if(dag.checkPipelineStatus(event, pipelineStatus)){
```

Comment:
The checkPipelineStatus method should exist in this class, not the DAG class. The same principle applies to the other classes in question